src/components/Timer.react.js

```diff
@@ -0,0 +1,322 @@
+     *  of milliseconds remaining.
+     *  When in stopwatch mode, the timer will count up from zero and show the number of milliseconds elapsed.
+     *  (read-only)
+     */
+    timer: PropTypes.number,
+
+    /**
+     * The timer will count down to zero in `countdown` mode and count up from zero in `stopwatch` mode
+     */
+    mode: PropTypes.oneOf(['stopwatch', 'countdown']),
+
+    /**
+     * Sets the number of milliseconds the timer will run.  If -1 the timer will not be limited by the duration. (the default)
+     * and if 0 then the timer stops running.
+     */
+    duration: PropTypes.number,
+
+    /**
+     * This will start the timer at the beginning with the given prop settings.
+     */
+    reset: PropTypes.bool,
+
+    /**
+     * A list of the time(s) in milliseconds at which to fire a callback. This can be used to start a task at a given
+     * time rather than using the timer. Since the timer is typically set at a small interval like one second, using
+     * fire_times can reduce the number of times a callback is fired and can increase app performance. The time(s) must be a
+     * multiple of the interval.
+     */
+    fire_times: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * This is updated when the timer reaches one of the times in the `fire_times` list. Using `at_fire_time` in a
+     * callback will trigger the callback at the time(s) in `fire_times` (Read only)
+     */
+    at_fire_time: PropTypes.number,
+
+    /**
+     * When True, the timer repeats once the timer has run for the number of milliseconds set in the duration.
+     */
+    rerun: PropTypes.bool,
+
+    /**
+     * Timer messages to be displayed by the component rather than showing the timer. It is a dictionary in the form of:
+     * {integer: string} where integer is the time in milliseconds of when the string message is to be displayed.
+     * Note: timer_format will override messages. For example, {10000 : "updating in 10 seconds"} will display the message
+     * "updating in 10 seconds" once the timer equals 10000.
+     */
+    messages: PropTypes.object,
+
+    /**
+     * If a timer is displayed, it will override timer `messages`.  This formats the timer (milliseconds) into human
+     * readable formats.  The options are:
+     *  `'none'`: no timer will be displayed;
+     *  `'display'`:  example - 1337000000 milliseconds will display as: '15d 11h 23m 20s';
+     *  `'compact'`: will show only the first unit: 1h 10m → 1h ;
+     *  `'verbose'`: will show full-length units. Example --  5 hours 1 minute 45 seconds
+     *  `'colonNotation'`: Useful when you want to show time without the time units, similar to
+     *                   a digital watch. Will always shows time in at least minutes: 1s → 0:01.
+     *                   Example - 5h 1m 45s → 5:01:45.
+     */
+    timer_format: PropTypes.oneOf([
+        'none',
+        'display',
```

Comment:
in `getFormat` it looks like this is `'default'`?

Reply:
:facepalm: How do you find stuff like this? (fixed now - thanks)